### PR TITLE
feat: create focus trap mixin and add it to modal

### PR DESCRIPTION
**Pull request by albrechtjohannes**
**Associated Issue**
#236 

**Summary**
This PR improves the accessibility of the modal component by adding a focus trap that is bound to the modal when opening it. To achieve this, I added a dependency to https://github.com/davidtheclark/focus-trap and a mixin, which is currently used only in the modal. However, it could also be used in other components, for example the popover component.

### CODE REVIEW

**Comment by albrechtjohannes** [comment]
For accessibility reasons we added a tabindex to the modal as well (e.g. for reading out the title). This As a result the browsers create a default outline when the modal gets focussed.

File: ui/src/mixins/FocusTrap.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import Vue from "vue";
+import createFocusTrap, { FocusTrap, Options } from "focus-trap";
+
+let focusTrap: FocusTrap;
```
Comment (ChristianKienle) [inline ui/src/mixins/FocusTrap.ts]: By having a single global `let focusTrap`-variable we are basically saying that there can only ever be a *one* focus trap active at a given time. Right? What if, further down the road, the focus trap mixin is also used in, let's say the Popover-component (which is used by the combobox-component) and the app has a combobox inside a modal?

My intuition says that the mixin needs non-reactive state:

```js
{
  created() {
    this.focusTrap = null; // maybe we should prefix this with fd?
  }
}
```

What do you think?

Comment (albrechtjohannes) [reply ui/src/mixins/FocusTrap.ts]: Yes, that's true. I will change it accordingly

Comment (albrechtjohannes) [reply ui/src/mixins/FocusTrap.ts]: I fixed it in the latest commit

**Pull request merged**